kitsune/tidings/events.py

```diff
@@ -108,8 +104,30 @@ class Event(object):
     #: ``set(['color', 'flavor'])``
     filters = set()
 
-    def fire(self, exclude=None, delay=True):
-        """Notify everyone watching the event.
+    @classmethod
+    def fire_async(cls, instance=None, exclude=None):
+        """
+        Enqueues a JSON-serializer-friendly Celery task that builds and sends emails
+        to everyone watching the event, excluding any users provided by "exclude". The
+        event will be constructed with the given instance if provided.
+        """
+        kwargs = {}
+        if instance:
+            kwargs.update(
+                instance_cls_module_name=instance.__class__.__module__,
+                instance_cls_name=instance.__class__.__name__,
+                instance_id=instance.id,
+            )
+        if exclude:
+            if not isinstance(exclude, Sequence):
+                exclude = [exclude]
+            kwargs.update(exclude_user_ids=[user.id for user in exclude])
```

Comment:
The `exclude` keyword argument can be either a single user object (that's why we check for a non-`Sequence`) or an iterable of user objects, but we need the list comprehension because we have to convert those user objects into JSON-friendly information that the Celery task accepts, so we convert them into a list of integers (ids).